core/foundation/inc/RtypesCore.h

```diff
@@ -34,91 +45,91 @@ class TRootIOCtor;
+typedef long           Long_t;      ///< Signed long integer 4 bytes (long). Size depends on architecture \deprecated Consider replacing with `long`
+typedef unsigned long  ULong_t;     ///< Unsigned long integer 4 bytes (unsigned long). Size depends on architecture \deprecated Consider replacing with `unsigned long`
 #endif
-typedef float          Float_t;     //Float 4 bytes (float)
-typedef float          Float16_t;   //Float 4 bytes written with a truncated mantissa
-typedef double         Double_t;    //Double 8 bytes
-typedef double         Double32_t;  //Double 8 bytes in memory, written as a 4 bytes float
-typedef long double    LongDouble_t;//Long Double
-typedef char           Text_t;      //General string (char)
-typedef bool           Bool_t;      //Boolean (0=false, 1=true) (bool)
-typedef unsigned char  Byte_t;      //Byte (8 bits) (unsigned char)
-typedef short          Version_t;   //Class version identifier (short)
-typedef const char     Option_t;    //Option string (const char)
-typedef int            Ssiz_t;      //String size (int)
-typedef float          Real_t;      //TVector and TMatrix element type (float)
-typedef long long          Long64_t; //Portable signed long integer 8 bytes
-typedef unsigned long long ULong64_t;//Portable unsigned long integer 8 bytes
+typedef float          Float_t;     ///< Float 4 bytes (float) \deprecated Consider replacing with `float`.
+typedef float          Float16_t;   ///< Float 4 bytes in memory, written to disk as 3 bytes (24-bits) by default or as a 4 bytes fixed-point-arithmetic Int_t (32-bits) if range was customized, with a truncated mantissa (12-bit by default in memory), and (7+1)-bits of exponent \warning Do not confuse Float16_t on file representation with a half-float such as std::float16_t.
+typedef double         Double_t;    ///< Double 8 bytes \deprecated Consider replacing with `double`.
+typedef double         Double32_t;  ///< Double 8 bytes in memory, written to disk as a 4 bytes Float_t (32-bits) by default, or as 3 bytes (24-bits) float if range is customized, with a truncated mantissa (24-bit by default in memory, less if range is customized), and (7+1)-bits of exponent \warning Do not confuse Double32_t on file representation with a single-precision float such as std::float32_t
+typedef long double    LongDouble_t;///< Long Double (not portable) \deprecated Consider replacing with `long double`.
+typedef char           Text_t;      ///< General string (char) \deprecated Consider replacing with `char`.
+typedef bool           Bool_t;      ///< Boolean (0=false, 1=true) (bool) \deprecated Consider replacing with `bool`.
+typedef unsigned char  Byte_t;      ///< Byte (8 bits) (unsigned char) \deprecated Consider replacing with `unsigned char` or `std::byte`.
+typedef short          Version_t;   ///< Class version identifier (short) \deprecated Consider replacing with `short` or `std::int16_t`.
```

Comment:
```suggestion
typedef short          Version_t;   ///< Class version identifier (short)
```
This is a semantic typedef.  (It matters that it is a Version Number, it does not matter (to the user) that it is a short).